File: front/src/types/Shift.ts
import { Nurse } from './Nurse';
import Qualification from './Qualification';

export type QueryShift = {
  id: string;
  startDate: Date;
  endDate: Date;
  nurseId: Nurse['id'];
  qualification: Qualification;
};

export type Shift = {
  id: string;
  startDate: Date;
  endDate: Date;
  nurse: Nurse | null;
  qualification: Qualification;
};

File: back/Shift.go
package main

type Shift struct {
	Id            string `json:"id"`
	StartDate     string `json:"startDate"`
	EndDate       string `json:"endDate"`
	NurseId       string `json:"nurseId"`
	Qualification string `json:"qualification"`
}

func NewShift(id, startDate, endDate, nurseId, qualification string) *Shift {
	return &Shift{
		Id:            id,
		StartDate:     startDate,
		EndDate:       endDate,
		NurseId:       nurseId,
		Qualification: qualification,
	}
}